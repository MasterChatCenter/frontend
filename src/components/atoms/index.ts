export { default as CardBorder } from './CardBorder';
export { default as AvatarUser } from './AvatarUser';
export { default as InputSearch } from './InputSearch';
