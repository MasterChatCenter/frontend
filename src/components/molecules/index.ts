export { default as AgentCard } from './AgentCard';
export { default as List } from './List';
