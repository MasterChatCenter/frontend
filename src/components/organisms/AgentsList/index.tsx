import React, { FC } from 'react';
import { AiOutlineUserAdd } from 'react-icons/ai';
import AgentCard from '@/molecules/AgentCard';
import { NotAgents, Div } from './styles';

type props = {
  users: any;
  onEdit: any;
  onDelete: any;
};

const AgentsList: FC<props> = ({ users, onEdit, onDelete }) => {
  if (users.length === 0) {
    return (
      <NotAgents>
        <p>No hay agentes agregados </p>
        <AiOutlineUserAdd />
      </NotAgents>
    );
  }

  return (
    <Div>
      {users.map((user: any, idx: number) => (
        <AgentCard
          key={idx}
          avatarUrl="https://www.tibs.org.tw/images/default.jpg"
          name={user.name + '' + user.lastname}
          nickName={user.name + '' + user.lastname}
          handleEdit={() => onEdit(user)}
          handleDelete={() => onDelete(user)}
        />
      ))}
    </Div>
  );
};

export default AgentsList;
