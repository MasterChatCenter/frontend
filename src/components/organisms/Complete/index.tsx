import React, { ReactNode } from 'react';
import { CardCenter } from './styles';
import CardShadow from '../../atoms/CardShadow/index';

type props = {
  children: ReactNode;
};

const Complete: React.FC<props> = ({ children }) => {
  return (
    <CardCenter>
      <CardShadow>
        <h2>Completar perfil</h2>
        {children}
      </CardShadow>
    </CardCenter>
  );
};

export default Complete;
