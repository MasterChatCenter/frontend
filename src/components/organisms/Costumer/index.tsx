import React, { FC } from 'react';
import InputEdit from '@/atoms/InputEdit';
import { CSSCostumer } from './styles';

const Costumer: FC = () => {
  return (
    <CSSCostumer>
      <InputEdit
        type="text"
        label="Nombre"
        placeholder="Escribe un nombre"
        value=""
      />
      <InputEdit
        type="email"
        label="Correo electronico"
        placeholder="Escribe un correo"
        value=""
      />
    </CSSCostumer>
  );
};

export default Costumer;
