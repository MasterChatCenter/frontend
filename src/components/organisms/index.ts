export { default as Register } from './Register';
export { default as Login } from './Login';
export { default as Complete } from './Complete';
export { default as ChatList } from './ChatList';
