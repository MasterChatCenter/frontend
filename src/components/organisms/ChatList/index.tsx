import React, { FC, useState, useEffect } from 'react';
import { InputSearch } from '@/atoms';
import { ChatButton, List } from '@/molecules';
import { CSSChatContainer, CSSWrapperSearch } from './styles';

type props = {
  allConversations: any;
  selectConversation: any;
};

const ChatList: FC<props> = ({ allConversations, selectConversation }) => {
  const [conversations, setConversations] = useState([...allConversations]);

  useEffect(() => {
    setConversations(allConversations);
  }, [allConversations]);

  const handleChange = (event: any) => {
    const value = event.target.value;
    const filter = allConversations.filter(
      ({ username, text }: any) =>
        username.indexOf(value) > -1 || text.indexOf(value) > -1
    );
    setConversations(filter);
  };

  const handleClick = (senderId: string) => {
    selectConversation(senderId);
  };

  return (
    <CSSChatContainer>
      <CSSWrapperSearch>
        <InputSearch handleChange={handleChange} />
      </CSSWrapperSearch>
      <List
        data={conversations}
        render={({ username, text, senderId }: any) => (
          <ChatButton
            key={senderId}
            image="/default-profile.jpg"
            username={username}
            message={text}
            handleClick={() => handleClick(senderId)}
          />
        )}
      />
    </CSSChatContainer>
  );
};

export default ChatList;
