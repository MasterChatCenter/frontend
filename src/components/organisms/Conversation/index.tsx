import React, { FC } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { InputChat } from '@/atoms';
import { List, Message } from '@/molecules';
import { sendMessageAction } from 'root/actions';
import { CSSConversation, CSSMessage, CSSInputChat } from './styles';

type props = {
  messages: any;
  senderId: string;
  conversationId: string | number;
};

const Conversation: FC<props> = ({ messages, senderId, conversationId }) => {
  const user = useSelector((store: any) => store.user);
  const dispatch = useDispatch();

  const onSend = (message: string) => {
    alert(message);
    const data = {
      messageData: {
        text: message,
        sendDate: '02-Sep-2020',
        senderId: senderId,
        conversation_id: conversationId,
      },
      senderId: senderId,
      tokenFacebook: user.company.tokenFacebook,
    };

    dispatch(sendMessageAction(data, `${user.name} ${user.lastname}`));
  };

  return (
    <CSSConversation>
      <List
        data={messages}
        render={({ username, text, type }: any, idx: number) => (
          <CSSMessage key={idx}>
            <Message
              type={type ? type : 'recipient'}
              username={username}
              text={text}
            />
          </CSSMessage>
        )}
      />
      <CSSInputChat>
        <InputChat onSend={onSend} />
      </CSSInputChat>
    </CSSConversation>
  );
};

export default Conversation;
