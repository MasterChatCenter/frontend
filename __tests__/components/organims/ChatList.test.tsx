import React from 'react';
import { ChatList } from '../../../src/components/organisms';
import { render } from '../../utils';

describe('organisms / ChatList', () => {
  const conversations = [
    {
      senderId: '123456',
      username: 'username',
      text: 'Menssage',
      id: 1,
    },
    {
      senderId: '654321',
      username: 'username',
      text: 'Menssage',
      id: 2,
    },
  ];

  it('matches snapshot', () => {
    const { asFragment } = render(
      <ChatList
        allConversations={conversations}
        selectConversation={() => alert('Selected')}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
