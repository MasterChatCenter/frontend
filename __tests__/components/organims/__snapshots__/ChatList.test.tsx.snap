// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms / ChatList matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="styles__CSSChatContainer-bad2t0-0 gbFroX"
  >
    <div
      class="styles__CSSWrapperSearch-bad2t0-1 hRrQFm"
    >
      <div
        class="styles__CSSChatBar-sc-1mrv7r9-0 bbBxmh"
      >
        <button
          class="styles__CSSChatAction-sc-1mrv7r9-2 kQeLNH"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
        </button>
        <input
          class="styles__CSSInput-sc-1mrv7r9-1 kmEPyU"
          placeholder="Escribe un mensaje"
        />
        <button
          class="styles__CSSChatAction-sc-1mrv7r9-2 kQeLNH"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div>
      <div
        class="styles__CSSCardBorder-sc-7hv695-0 kweAYS"
      >
        <div
          class="styles__CSSContainer-sc-1kwkyv1-2 hLOejI"
        >
          <div
            class="styles__CSSDivider-sc-1kwkyv1-5 MbLtO"
          >
            <img
              alt="username"
              class="styles__AvatarUserStyle-y86xw5-0 bQhvet"
              src="/default-profile.jpg"
            />
            <div>
              <h2
                class="styles__CSSNickName-sc-1kwkyv1-1 cyocWw"
              >
                username
              </h2>
              <p>
                Menssage
              </p>
            </div>
          </div>
          <div
            class="styles__CSSStatusRed-sc-1kwkyv1-4 kuSMtD"
          />
        </div>
      </div>
      <div
        class="styles__CSSCardBorder-sc-7hv695-0 kweAYS"
      >
        <div
          class="styles__CSSContainer-sc-1kwkyv1-2 hLOejI"
        >
          <div
            class="styles__CSSDivider-sc-1kwkyv1-5 MbLtO"
          >
            <img
              alt="username"
              class="styles__AvatarUserStyle-y86xw5-0 bQhvet"
              src="/default-profile.jpg"
            />
            <div>
              <h2
                class="styles__CSSNickName-sc-1kwkyv1-1 cyocWw"
              >
                username
              </h2>
              <p>
                Menssage
              </p>
            </div>
          </div>
          <div
            class="styles__CSSStatusRed-sc-1kwkyv1-4 kuSMtD"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
